test/e2e/upgrades/apps/job.go

```diff
@@ -70,7 +73,7 @@ func (t *JobUpgradeTest) Teardown(ctx context.Context, f *framework.Framework) {
 	// rely on the namespace deletion to clean up everything
 }
 
-// ensureAllJobPodsRunning uses c to check in the Job named jobName in ns
+// ensureAllJobPodsRunning uses c to check if the Job named jobName in ns
```

Comment:
I know this is an old function, but I'm wondering how do we know the pods are recreated already (and running)? I'm wondering if we are missing a wait here, but even if so this is a separate issue.

Reply:
This is being already handled at the upgrade framework level (see here: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/upgrades/upgrade_suite.go). These jobs here are implementing `upgrades.Test` interface, so you only focus on the `Setup` (create the job and make sure it's running) and `Test` (check the job after the upgrade) steps.